src/components/instagram-profile.tsx

```diff
@@ -0,0 +1,245 @@
+            {profile.posts.map((post, index) => (
+              <div
+                key={index}
+                className="aspect-square hover:scale-105 hover:shadow-2xl transition-all cursor-pointer"
+                onClick={() => setCurrentPost(post)}
+              >
+                <Image
+                  src={post.imageSrc}
+                  alt={post.caption}
+                  width="200"
+                  height="200"
+                  className="w-full h-full object-cover"
+                />
+              </div>
+            ))}
+          </section>
+        )}
+      </div>
+    </div>
+  );
+}
+
+function VerticalNumberText({
+  number,
+  text,
+  onClick,
+}: {
+  number: number;
+  text: string;
+  onClick?: () => void;
+}) {
+  return (
+    <div
+      className="inline-flex flex-col items-center hover:scale-105 transition-all cursor-pointer"
+      onClick={onClick}
+    >
+      <span className="font-semibold text-xl">{number}</span>
+      <span className="font-light">{text}</span>
+    </div>
+  );
+}
+
+function Following({
+  following,
+  onClick,
+}: {
+  following: string[];
+  onClick: () => void;
+}) {
+  return (
+    <div className="w-full h-full z-30 flex items-center absolute">
+      <div className="p-5 z-30 h-full w-full flex items-center">
+        <div className="bg-white p-2 pb-4 rounded-xl w-full h-full">
+          <header className="flex p-2 justify-between">
+            <span className="font-semibold">Following</span>
+            <div
+              onClick={onClick}
+              className="hover:scale-105 transition-all cursor-pointer"
+            >
+              <CloseIcon />
+            </div>
+          </header>
+          <div className="flex flex-col px-4 pt-4 overflow-y-auto max-h-[90%]">
+            {following.map((follower) => (
+              <div
+                key={follower}
+                className="flex items-center py-4 border-b-[1px] last:border-0"
+              >
+                {/*<Image*/}
+                {/*  src={"/passwort.png"} alt={"Profile picture"} width="50" height="50"/>*/}
+                <div className="rounded-full border-[1px] border-gray-300 w-8 h-8 flex-shrink-0 bg-white mr-4 flex items-center justify-center">
+                  <User weight="fill" className="opacity-30" />
+                </div>
+                <span>{follower}</span>
+              </div>
+            ))}
+            <span className="italic opacity-50 p-4 inline-block">
+              Weitere Konten konnten nicht geladen werden.
+            </span>
+          </div>
+        </div>
+      </div>
+      <div className="absolute bg-black opacity-40 w-full h-full top-0"></div>
+    </div>
+  );
+}
+
+function CurrentPost({
+  post,
+  profile,
+  onClick,
+}: {
+  post: InstagramPost;
+  profile: InstagramProfileData;
+  onClick: () => void;
+}) {
+  return (
+    <div
+      className="w-full h-full z-30 flex items-center absolute cursor-pointer"
+      onClick={onClick}
```

Comment:
use another tag. not the div tagö

Reply:
instagram profil hab ich 1 zu 1 von jonas übernommen aus dem modul datenverarbeitung. Können wir am besten direkt bei ihm ändern